### DISCUSSION TITLE
Feature request: redesign the scripting/rules interface

### DISCUSSION THREAD

**Discussion by oxygenic** [arendst/Tasmota#16907] [Ideas]
I recently tried to do something wiht Tasmota which - from my experience - should be incredibly simply: reading one single value out of a Tasmota device and send it via HTTP GET to a server once a day.

Unfortunately it was not that easy: there are a lot of very fine details about rules and commands available in the online-documentation, but some useful examples and an overview to get the whole picture and a general understanding is missing completely.
When asking here in the webboard (https://github.com/arendst/Tasmota/discussions/16877), it becomes obvious, nobody really understands these rules, so it was impossible to get some answers that at least could be used as starting point.

So to sum it up: may be a few people understand the rules stuff, but it is not user-friendly enough, to be used by everybody.

Thus my feature-request which in my opinion would bring a big improvement: redesign the whole scripting/rules stuff to be more straight-forward, be less hardware-dependent and be easier to understand and to use.

**Comment by sfromis** [comment]
What I see in that thread is not lack of understanding of how rules work, but lack of clarity of what your specific situation is, and communication difficulty about how words are used. Use of rule triggers like Energy#Yesterday is simply a case of JSON-based rule triggers described in the rule docs page. But yeah, rules definitely have a learning curve, you should expect to have to play around with simpler examples to grok how it works.

**Reply by oxygenic** [reply]
What exactly is simpler than sending one simple value to an URL via a HTTP-GET-request?

**Reply by barbudor** [reply]
What is more simpler than using a time-series database such as influxdb or prometheus that are natively supported instead of Sql? 

Rules are simple, basic, right what works fine on simple esp8266
Of course you can take a raspberry pi with node red and that would be a simple flow with 2 nodes

On Tasmota you need 1 rule to extract the value you want and one rule to publish it at the time of day you want. The complexity of needing 2 rules is because you want to publish only once a day. 
Publishing at each teleperiod would be only 1 rule

Rules have been working fine for everyone for simple use case like yours. Those wanting more can use Scripting-Language (more powerful but more complex) or get an ESP32 and use Berry language 

We won't redesign the rule engine but you can suggest focused improvements based on your recent experience.

**Reply by oxygenic** [reply]
Sorry for the harsh words, but sometimes I have the feeling you're jerking me around:

> On Tasmota you need 1 rule to extract the value you want and one rule to publish it at the time of day you want.

So what would this rule look like? I have asked this question in the other thread but got no answer that would be helpful in any way!

> The complexity of needing 2 rules is because you want to publish only once a day. Publishing at each teleperiod would be only 1 rule

So what would these two rules look like? I have asked this question in the other thread but got no answer that would be helpful in any way!

> Rules have been working fine for everyone for simple use case like yours.

So what would the rule for my simple use case look like? I have asked this question in the other thread but got no answer that would be helpful in any way!

And please do not start again with your "it is all in the docs"-answers. As stated several times in the other thread, what you claimed to be in the docs was NOT there.

**Reply by pkkrusty** [reply]
From the docs: https://tasmota.github.io/docs/Rules/#publish-maximum-value-from-sensor-in-a-time-period Is a pretty good starting point for your requirement.  Seems like some of your frustration is coming from the fact that you have issues with your device (i.e. not seeing anything in the console). How did Tasmota get on it in the first place? Did it come pre-flashed? Can you put MQTT on your remote server? That would simplify things, since Tasmota and MQTT are well integrated, then processing into the SQL database can happen on the remote machine.

**Reply by oxygenic** [reply]
Your link describes MQTT-stuff. All I wanted to do is a plain HTTP GET call with a the yesterdays energy value transmitted as parameter. No extensive MQTT-broker installation/configuration. I have found a lot of such documentation-slivers which describe some single, specific parts very detailed but do not help to get the whole picture in any way.

Please refer to the original thread: I have given up Tasmota as I'm not willing to waste even more time. The feature requests here are intended to show where problems are and what can be improved.

Feel free to use it as helpful information to find a more user-friendly solution or to ingore it and to stay with the current, over-complex stuff which can be used by very few people only, for me it does not matter any more.

**Reply by meingraham** [reply]
@oxygenic - Good luck & Bon Voyage!

**Reply by arendst** [reply]
Ah Michael, missed you for a long while! Great to be back.
